app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/recordaudio/RecordAudioViewModel.kt

```diff
@@ -0,0 +1,323 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.android.ui.home.messagecomposer.recordaudio
+
+import android.content.Context
+import android.media.MediaRecorder
+import android.os.Build
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.core.net.toUri
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.wire.android.appLogger
+import com.wire.android.media.audiomessage.AudioState
+import com.wire.android.media.audiomessage.RecordAudioMessagePlayer
+import com.wire.android.ui.home.conversations.model.UriAsset
+import com.wire.android.util.CurrentScreen
+import com.wire.android.util.CurrentScreenManager
+import com.wire.android.util.ui.UIText
+import com.wire.kalium.logic.data.asset.KaliumFileSystem
+import com.wire.kalium.logic.feature.asset.GetAssetSizeLimitUseCase
+import com.wire.kalium.logic.feature.call.usecase.ObserveEstablishedCallsUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.launch
+import java.io.File
+import java.io.IOException
+import javax.inject.Inject
+import kotlin.io.path.deleteIfExists
+import kotlin.properties.Delegates
+
+@Suppress("TooManyFunctions")
+interface RecordAudioViewModel {
+    fun getButtonState(): RecordAudioButtonState
+    fun getDiscardDialogState(): RecordAudioDialogState
+    fun getPermissionsDeniedDialogState(): RecordAudioDialogState
+    fun getMaxFileSizeReachedDialogState(): RecordAudioDialogState
+    fun getInfoMessage(): SharedFlow<UIText>
+    fun getOutputFile(): File?
+    fun getAudioState(): AudioState
+    fun startRecording(context: Context)
+    fun stopRecording()
+    fun showDiscardRecordingDialog(onCloseRecordAudio: () -> Unit)
+    fun onDismissDiscardDialog()
+    fun showPermissionsDeniedDialog()
+    fun onDismissPermissionsDeniedDialog()
+    fun onDismissMaxFileSizeReachedDialog()
+    fun discardRecording(onCloseRecordAudio: () -> Unit)
+    fun sendRecording(onAudioRecorded: (UriAsset) -> Unit, onComplete: () -> Unit)
+    fun onPlayAudio()
+    fun onSliderPositionChange(position: Int)
+}
```

Comment:
Just wondering why it's interface ?

Reply:
No specific reason, guess its just better for testing later on so I just went along with it, do you see any issue with it?

Reply:
No issue, I just don't see any reason of using interface like multiple Inheritance or total abstraction..

It's more lines..

and for testing we are already testing viewModel without using interface, it can't be an issue I guess